addons/pinniped/config-controller/controllers/controller_test.go

```diff
@@ -42,215 +44,166 @@ var _ = Describe("Controller", func() {
+					constants.TKGClusterNameLabel: cluster.Name,
+				},
+			},
+			Type: constants.ClusterBootstrapManagedSecret,
+		}
 	})
 
-	Context("Cluster", func() {
+	Context("ClusterBootstrap Secret", func() {
 		BeforeEach(func() {
 			create(ctx, cluster)
+			create(ctx, pinnipedCBSecret)
 		})
 
 		AfterEach(func() {
-			delete(ctx, cluster)
-		})
-
-		When("cluster is created", func() {
-			It("creates a secret with identity_management_type set to none", func() {
-				Eventually(addonSecretFunc(ctx, cluster, nil)).Should(Succeed())
-			})
+			deleteObject(ctx, cluster)
+			deleteObject(ctx, pinnipedCBSecret)
 		})
 
-		When("cluster is updated", func() {
-			BeforeEach(func() {
-				clusterCopy := cluster.DeepCopy()
-				annotations := clusterCopy.ObjectMeta.Annotations
-				if annotations == nil {
-					annotations = make(map[string]string)
-				}
-				annotations["sweetest-cat"] = "lionel"
-				clusterCopy.ObjectMeta.Annotations = annotations
-				Expect(k8sClient.Update(ctx, clusterCopy)).To(Succeed())
-			})
-			// TODO: test where we edit that TKR label on the cluster.....................................
-			It("secret remains unchanged", func() {
-				Eventually(addonSecretFunc(ctx, cluster, nil)).Should(Succeed())
+		When("the secret gets created", func() {
+			It("updates the secret with the proper data values", func() {
+				Eventually(cbSecretFunc(ctx, cluster, nil)).Should(Succeed())
 			})
 		})
 
-		When("Cluster is deleted", func() {
+		When("the secret gets updated", func() {
 			BeforeEach(func() {
-				delete(ctx, cluster)
-			})
-
-			It("deletes the Pinniped addon secret associated with the cluster", func() {
 				Eventually(func(g Gomega) {
-					gotSecret := &corev1.Secret{
-						ObjectMeta: metav1.ObjectMeta{
-							Namespace: pinnipedNamespace,
-							Name:      fmt.Sprintf("%s-pinniped-addon", cluster.Name),
-						},
-					}
-					err := k8sClient.Get(ctx, client.ObjectKeyFromObject(gotSecret), gotSecret)
-					g.Expect(k8serrors.IsNotFound(err)).To(BeTrue())
+					err := k8sClient.Get(ctx, client.ObjectKeyFromObject(pinnipedCBSecret), pinnipedCBSecret)
+					g.Expect(err).NotTo(HaveOccurred())
 				}).Should(Succeed())
+				secretCopy := pinnipedCBSecret.DeepCopy()
+				updatedSecretDataValues := map[string][]byte{
+					constants.TKGDataValueFieldName: []byte(fmt.Sprintf("%s: fire", constants.IdentityManagementTypeKey)),
+				}
+				secretCopy.Data = updatedSecretDataValues
+				Expect(k8sClient.Update(ctx, secretCopy)).To(Succeed())
 			})
-		})
-	})
 
-	Context("Addon Secret", func() {
-		BeforeEach(func() {
-			create(ctx, cluster)
-		})
-
-		AfterEach(func() {
-			delete(ctx, cluster)
+			It("updates the secret with the proper data values", func() {
+				Eventually(cbSecretFunc(ctx, cluster, nil)).Should(Succeed())
+			})
 		})
 
-		When("the secret gets deleted", func() {
-			var gotSecret *corev1.Secret
-
+		When("random values are added to the secret", func() {
```

Comment:
praise: i really like this test

Reply:
in the future, we should add a test that makes sure that if someone has set an `overlay.yaml` in the `Secret`, that is preserved

Reply:
oooh good thought.  I'll add that now.